tool/main.re

```diff
@@ -246,22 +247,152 @@ static MULTIPLAYER_MENU = Ui::new("Multiplayer:", List::of(
     }),
 ));
 
+static mut SAVE_RECORDING_LABEL = Text { text: "Save Recording" };
+static mut RECORDING_NAME_LABEL = Text { text: "Recording name" };
+static mut LIST_RECORDINGS_LABEL = Text { text: "List Recordings" };
+static mut DELETE_RECORDING_LABEL = Text { text: "Delete Recording" };
+
+fn main_list_recordings() {
+    let recordings_list = Tas::list_recordings();
+    LIST_RECORDINGS_LABEL.text = f"List Recordings ({recordings_list})";
+}
+
 static UTIL_MENU = Ui::new("Util:", List::of(
     UiElement::Input(Input {
-        label: Text { text: "Save Recording" },
-        input: "",
+        label: Text { text: "Player Name" },
+        input: SETTINGS.multiplayer_name,
         onclick: fn(input: string) {
-            tas_save_recording(input);
+            SETTINGS.set_multiplayer_name(input);
         },
         onchange: fn(input: string) {},
     }),
-    UiElement::Input(Input {
+    UiElement::Button(UiButton {
+        label: Text { text: "Save Recording" },
+        onclick: fn(label: Text) {
+            RECORDING_NAME_LABEL.text = f"Recording name";
+            let recordings_list = Tas::list_recordings();
+            let mut recordings = List::new();
+            recordings.push(UiElement::Input(Input {
+                label: RECORDING_NAME_LABEL,
+                input: "",
+                onclick: fn(input: string) {
+                    let recordings_list = Tas::list_recordings();
+                    if input.len_utf8() == 0 {
+                        RECORDING_NAME_LABEL.text = f"Recording name (Error: empty name)";
+                        return;
+                    }
+                    RECORDING_NAME_LABEL.text = f"Recording name";
+                    tas_save_recording(input);
+                },
+                onchange: fn(input: string) {}
+            }));
+            for recording in recordings_list {
+                recordings.push(UiElement::Button(UiButton {
+                    label: Text { text: recording },
+                    onclick: fn(label: Text) {
+                        tas_save_recording(label.text);
+                    },
+                }));
+            }
+            recordings.push(UiElement::Button(UiButton {
+                label: Text { text: "Back" },
+                onclick: fn(label: Text) { leave_ui() },
+            }));
```

Comment:
This is the one time I'd put the "Back" button at the beginning. Otherwise it may not be visible on lower resolution screens.